### ISSUE TITLE
Explicit export?

### ISSUE THREAD

**Issue by lathonez** [szimek/signature_pad#240]
Hi,

I help to maintain https://github.com/wulfsolter/angular2-signaturepad, which is a simple ng2 component wrapper for this repo.

We've been asked to upgrade to take your latest 2.0.0 (https://github.com/wulfsolter/angular2-signaturepad/issues/30), but I am unable to compile the project as it uses a default export: https://github.com/angular/angular/issues/11402

At the moment I'm unsure if you have an alternative option, which is why the issue is phrased like this:

Could you instead (somehow) explicitly export` SignaturePad`, so we're able to use AOT compiling to build it with Angular 2? 

Thanks for the excellent work here.

Stephen

**Comment by szimek** [comment]
@lathonez I'll try to look into it, though it might not be that easy...

If I try to export SignaturePad as the default _and_ named export, I get the following warning from rollup:
> Using named and default exports together. Consumers of your bundle will have to use `SignaturePad['default']` to access the default export, which may not be what you want. Use `exports: 'named'` to disable this warning

If I'd export only named export, everyone would have to use `import { SignaturePad } from 'signature_pad';`, which is strange, as it's the only exported property and I'd have to release it as 3.0, because it would be a breaking change :/

Any ideas how to solve it are welcome.

**Comment by lathonez** [comment]
Thanks for such a quick response.

The latter (export only named export), is the standard in my experience - also what the team are recommending on the AOT issue. I appreciate that it doesn't help you with the semver issue.

Presumably the former would also require a breaking change as it appears to change the default access path.

The only other thing I can think of is doing something dirty with an additional class:

* create some other class that imports `SignaturePad` as default
* export a named export (as some other class name `SignaturePadNamed`) from there so we can use it for AOT

However:

* I'm unsure if this works in principle
* Equally unsure as to whether it solves the AOT problem (e.g we'd still be depending on something with a default export upstream)
* I think it's nasty 😞 

If you're keen to look at the above as a workaround I'm very happy to help test it.

My suggestion though, would be to confirm whether or not the named export is the way to go, and just take the hit on semver if it is.

**Comment by DynamoEffects** [comment]
+1ing this, we're in urgent need of the SVG feature in 1.6.0+.  In the meantime we'll just fork both projects, but just wanted to let you know that this is important to us and hope you can find a solution that works for everyone.

**Comment by lathonez** [comment]
@szimek FYI @DynamoEffects has got a working fork of both.

https://github.com/wulfsolter/angular2-signaturepad/issues/30#issuecomment-298843373. AFAICS it was just removing the default export

**Comment by lathonez** [comment]
It wasn't enough, he took 1.5.3 and ported some fixes he needed. This is still on my list, just don't have time at the moment

**Comment by lathonez** [comment]
Hey,

I've been able to solve this (thanks to @DynamoEffects) by using bit of a shim around require:

```javascript
 let sp: any = require('signature_pad')['default'];
```

https://github.com/wulfsolter/angular2-signaturepad/blob/2e5b54a33e703d54e159f7c64e1d8d7e081d544f/signature-pad.ts#L36

I'm sure this still isn't the ideal / recommended way. I tried lots of variations on `import { SignaturePad }` whilst making it an explicit export, (as opposed to using require), but could never get it to compile.

I think you'll want to address this for correctness / compatibility with AOT project going forward, but from my perspective I am happy if you want to close it - so long as we'll be able to keep using this require shim in future.

Thanks.
